### ISSUE TITLE
Ports disappear for a minute leading to huge spikes in traffic graphs

### ISSUE THREAD

**Issue by Freeaqingme** [librenms/librenms#14789]
### The problem

For some reason some of my port data disappears from LibreNMS and is added back a minute later. I'm not sure why this is happening (a device issue, no doubt). That's not a problem in and of itself. However, the traffic graphs show a huge spike the minute it's added back.

Would it be possible to think of a way to prevent these spikes from happening?

![Screenshot from 2023-01-17 13-46-23](https://user-images.githubusercontent.com/33034/212903744-d408f572-e12c-4a4c-afc6-e688e1eae0a1.png)
![Screenshot from 2023-01-17 13-46-05](https://user-images.githubusercontent.com/33034/212903747-3abd8f81-2901-4d81-bb4e-38746c4c73e9.png)
![Screenshot from 2023-01-17 13-44-46](https://user-images.githubusercontent.com/33034/212903751-6c1c1ecd-e923-44c7-b60e-59888a838333.png)

### Output of ./validate.php

```txt
librenms@librenms1:~$ ./validate.php 
===========================================
Component | Version
--------- | -------
LibreNMS  | 22.12.0-16-g437479382 (2023-01-12T08:13:08+00:00)
DB Schema | 2022_09_03_091314_update_ports_adsl_table_with_defaults (248)
PHP       | 8.2.1
Python    | 3.8.10
Database  | MariaDB 10.3.37-MariaDB-0ubuntu0.20.04.1
RRDTool   | 1.7.2
SNMP      | 5.8
===========================================

[OK]    Composer Version: 2.5.1
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    
[OK]    Database schema correct
[OK]    MySQl and PHP time match
[OK]    Active pollers found
[OK]    Dispatcher Service not detected
[OK]    Locks are functional
[OK]    Python poller wrapper is polling
[OK]    Redis is unavailable
[OK]    rrd_dir is writable
[OK]    rrdtool version ok
[WARN]  Your install is over 24 hours out of date, last update: Thu, 12 Jan 2023 08:13:08 +0000
	[FIX]: 
	Make sure your daily.sh cron is running and run ./daily.sh by hand to see if there are any errors.
[WARN]  Your local git contains modified files, this could prevent automatic updates.
	[FIX]: 
	You can fix this with ./scripts/github-remove
	Modified Files:
	 includes/definitions/discovery/eatonpdu.yaml
	 includes/definitions/eatonpdu.yaml
	 mibs/eaton/EATON-EPDU-MIB
[FAIL]  We have found some files that are owned by a different user than 'librenms', this will stop you updating automatically and / or rrd files being updated causing graphs to fail.
	[FIX]: 
	sudo chown -R librenms:librenms /opt/librenms
	sudo setfacl -d -m g::rwx /opt/librenms/rrd /opt/librenms/logs /opt/librenms/bootstrap/cache/ /opt/librenms/storage/
	sudo chmod -R ug=rwX /opt/librenms/rrd /opt/librenms/logs /opt/librenms/bootstrap/cache/ /opt/librenms/storage/
	Files:
	 /opt/librenms/html
	 /opt/librenms/html/api_v0.php
	 /opt/librenms/html/js
	 /opt/librenms/html/js/maps
	 /opt/librenms/html/js/maps/france_departments.js
	 /opt/librenms/html/js/maps/world_countries.js
	 /opt/librenms/html/js/maps/usa_states.js
	 /opt/librenms/html/js/popper.min.js.map
	 /opt/librenms/html/js/bootstrap4.min.js
	 /opt/librenms/html/js/boot.js
	 /opt/librenms/html/js/register-service-worker.js
	 /opt/librenms/html/js/justgage.min.js.map
	 /opt/librenms/html/js/typeahead.bundle.min.js
	 /opt/librenms/html/js/app.js
	 /opt/librenms/html/js/bootstrap-hover-dropdown.min.js
	  and 977 more...
librenms@librenms1:~$
```

### What was the last working version of LibreNMS?

The same issue was happening while running on a version dating a year back

**Comment by Freeaqingme** [comment]
I just found RRDTune, that could possibly mitigate this issue. Will monitor if this helps.

https://docs.librenms.org/Extensions/RRDTune/

**Comment by Freeaqingme** [comment]
So the RRDtune setting appears to fix everything on the side of Librenms. Now only to figure out how to get Mikrotik in order to fix the actual cause of this issue. That's not for this ticket, however.

**Comment by Freeaqingme** [comment]
For those finding this issue through Google: This issue started occurring after adding a bunch of new peers (and routes). It turned out that the 'netstats' module was taking 50s+ for every poll action. Disabling this module fixed everything.

**Comment by murrant** [comment]
@Freeaqingme that was fixed in 23.1.1
